# Trajectory-Inference-and-Genes-Analysis-in-Ionocytes-under-the-context-of-Cystic-Fibrosis

## Introduction
Many interesting biological processes could demonstrate itself as a continuum of dynamic cellular states. In  Part 1 and Part 2, we mainly focused on analyzing the clustered ionocytes and their subtypes, which are static snapshots of discrete cellular states and could not reveal temporal changes in gene expression underlying dynamic biological processes such as cell differentiation and disease progression. In this module, I will perform pseudotime analysis to study the dynamic properties of human airway epithelial cells, with a focus on ionocytes and their subtypes. By analyzing their differential expressed genes with respect to pseudotime and the enriched pathways, the main goal here is to explore potential ionocyte cell state transitions that might be associated with cystic fibrosis, providing insight into potential therapeutic targets.

## Data and methods
### Data: Clustered human airway epithelium single cell data from normal and cystic fibrosis lungs
The source data comes from a multi-institute consortium studying human airway epithelium at the single-cell level, encompassing 19 donors with cystic fibrosis undergoing transplantation for end-stage lung disease and 19 donors with previously healthy lungs. The preprocessed data after quality control, expression normalization, and batch effect correction was downloaded from GEO database in the format of rds (data/GSE150674_Seurat_Object_ALI.rds) using accession number GSE150674 [1]. The data contains 3000 genes and 10706 cells. In previous analysis modules, five clusters were identified, with cluster 4 being identified as ionocytes (n =121, CF:CO=58:63). Further clustering analysis in the ionocytes found 2 subtypes, with 74 cells belonging to subtype 0 (CF:CO= 39 :35) and 47 cells belonging to subtype 1 (CF:CO=28:19). I will start from the clustered data for all cells (data/GEO1ClusterSeurat.rds) and clustered ionocytes data (data/ClusteredIono.rds) to perform trajectory analysis in terms of airway epithelial cells, ionocytes, and ionocyte subtypes.

### Method: Trajectory inference with Slingshot 2.10.0
Trajectory inference or pseudotemporal ordering is a computational method to study cells that change their states by undergoing gradual transcriptional changes, with progress being driven by an underlying temporal variable or pseudotime.  I used an R package named “Slingshot”, which is an unsupervised learning method that can stably reconstruct the pseudo-temporal order by first inferring global lineage structure with cluster-based minimum spanning tree and then inferring pseudotime variables for cells along each lineage with simultaneous principal curves [12]. I used Slingshot to perform trajectory inferences using the clustered results based on t-SNE in three levels:  all cells, ionocytes and ionocyte subtypes (Fig 3A, Fig 3B, Fig 3E & Fig 3F). To explore the potential relationship between dynamic cell states and Cystic Fibrosis disease, I performed the nonparametric two-sample Kolmogorov-Simrnov test to quantitatively evaluate whether there is a significant difference between the distribution of CF vs CO along the trajectory. 

### Method: Trajectory differential genes analysis with TSCAN 1.40.1
I used the testPseudotime function in the R package TSCAN to detect differentially expressed genes given a constructed pseudo time ordering by fitting a natural spline and using ANOVA to test whether any of the spline coefficients are non-zero [13]. Using a false discovery rate of 0.05 and an absolute log fold change of 0.05 (suggested by the paper), I found 93 differentially expressed genes in the ionocyte trajectory (data/Ionocyte_Trajectory_genes.csv), 60 differentially expressed genes in the ionocyte subtype 0 trajectory (data/Ionocyte_Trajectory_s0_genes.csv), and 136 differentially expressed genes in ionocyte subtype 1 trajectory respectively (data/Ionocyte_Trajectory_s1_genes.csv). The expression of these genes along the estimated pseudotime in ionocytes, ionocyte subtype 0 and ionocyte subtype 1 is visualized as heat maps respectively (Fig 3D, Fig 3G && Fig 3H).

### Method: Trajectory gene set enrichment analysis with fgsea 1.28.0 
To interpret the biological functions of the trajectory differentially expressed genes, I performed gene set enrichment analysis using the R package fgsea [14]. I ranked all the genes using the metric sign(log2 fold change)*(-log10 p value), in which the magnitude of the ranking will be given by the p-value, and the sign of the ranking will be given by the fold-change. The background I used comes from the Molecular Signatures Database (MSigDB) [15]. I used human biological pathway (BP) subcollection in the C5 collection (ontology gene sets) to perform enrichment analysis for differentially expressed genes in ionocyte trajectory and ionocyte subtypes trajectory, with the  background genes filtered to the same 3000 genes as our data. Using FDR<0.05, no statistically significant enriched pathways are found for trajectory genes in ionocyte subtypes. The enriched pathway for overall ionocyte trajectory genes are shown in Fig 3C.


### Results and discussion
From the global trajectory reconstruction results using all human airway cells in our dataset (Fig 3A), we can see two estimated lineages. Both estimated lineage trajectories demonstrated a continuous trajectory along Cluster 3-Cluster 1-Cluster 0. The first lineage then connects Cluster 2, while the second lineage then traverses through Cluster 4, which is the ionocyte cluster of our interest. Based on the cell type annotation (Cluster 0: Basal cells, Cluster 1: Basal cells, Cluster 2: Secretory cells), it is likely that the shared trajectory recapitulates the known basal-to-secretory cell lineage, in which basal cells are a recognized stem cell population in both human and mouse lungs [16]. This finding holds significant promise for informing ionocyte cell therapy proposed recently for cystic fibrosis [17], as it delineates a potential differentiation path towards ionocytes.
Focusing on the trajectory of ionocyte and its two subtypes, one interesting finding is that CF and CO ionocyte in subtype 0 shows statistically significant distribution over the estimated pseudo time (two-sample Kolmogorov-Simrnov test, D=0.44, p-value = 8.86e-4), which might indicate potential relationship between cell state transition and CF in ionocyte subtype 0 (Fig 3G). The reasonability of this finding is further supported when we look at the differential genes (data/Ionocyte_Trajectory_s0_genes.csv). For example, the differentially expressed genes CLCNKB (logFC=-0.12, FDR=0.0056), CLIC6 (logFC=0.06, FDR=6.4e-10) are related to chloride channels, which is important for maintaining chloride ion balance [18], and abnormal chloride balance could related to cystic fibrosis. Besides, other differentially expressed genes such as SCGB3A1 (logFC=0.13, FDR=1.35e-10) and PIGR (logFC=0.06,FDR=6.26e-8) are related to airway epithelial immune regulation [18], and previous researches has indicated potential roles of immunity with cystic fibrosis [19]. However, no statistical significant enrichment of pathways is found for these genes.
There are enriched pathways for differentially expressed genes along the pseudo time in the overall ionocytes trajectory. From Fig 3D, we could observe 3 clustered expression patterns of these genes along the estimated pseudotime . Based on the gene set enrichment analysis, one interesting finding is that ion homeostasis and ion transmembrane transport are identified as enriched pathways (Fig 3C), suggesting a transition of cell states regarding ion-regulation in ionocytes. This could be important since cystitis fibrosis is associated with abnormal ion regulation, and ionocyte is a potential target for therapeutic intervention. Understanding this cell state transition could offer us insight into critical time points for therapeutics target.
Overall, this part mainly discussed the potential cell state transition in ionocytes, and providing potential insight for the pathology and therapy for cystic fibrosis. Instead of focusing on single causal genes such as CFTR in ionocytes, context-based cell state might also affect cystic fibrosis, which could be a potential reason why CFTR gene expression alone is not enough to explain the cystitis fibrosis.

### Reference
1. Carraro, G., Langerman, J., Sabri, S. et al. Transcriptional analysis of cystic fibrosis airways at single-cell resolution reveals altered epithelial cell states and composition. Nat Med 27, 806–814 (2021). https://doi.org/10.1038/s41591-021-01332-7
2. Montoro DT, Haber AL, Biton M, Vinarsky V, Lin B, Birket SE, Yuan F, Chen S, Leung HM, Villoria J, Rogel N, Burgin G, Tsankov AM, Waghray A, Slyper M, Waldman J, Nguyen L, Dionne D, Rozenblatt-Rosen O, Tata PR, Mou H, Shivaraju M, Bihler H, Mense M, Tearney GJ, Rowe  SM, Engelhardt JF, Regev A, Rajagopal J. A revised airway epithelial hierarchy includes CFTR-expressing ionocytes. Nature. 2018 Aug;560(7718):319-324. doi: 10.1038/s41586-018-0393-7. Epub 2018 Aug 1. PMID: 30069044; PMCID: PMC6295155.
3. Plasschaert LW, Žilionis R, Choo-Wing R, Savova V, Knehr J, Roma G, Klein AM, Jaffe AB. A single-cell atlas of the airway epithelium reveals the CFTR-rich pulmonary ionocyte. Nature. 2018 Aug;560(7718):377-381. doi: 10.1038/s41586-018-0394-6. Epub 2018 Aug 1. PMID: 30069046; PMCID: PMC6108322.
4. Carraro, G., Langerman, J., Sabri, S. et al. Transcriptional analysis of cystic fibrosis airways at single-cell resolution reveals altered epithelial cell states and composition. Nat Med 27, 806–814 (2021). https://doi.org/10.1038/s41591-021-01332-7
5. Kumar, Shankar & Tana, Anand & Shankar, Anu. (2014). Cystic fibrosis—what are the prospects for a cure? Eur J Intern Med. European Journal of Internal Medicine. 25. 803-07. 10.1016/j.ejim.2014.09.018. 
6. Shah, Lin, Waghray, Rajagopal. et al. Cystic Fibrosis and the Cells of the Airway Epithelium: What Are Ionocytes and What Do They Do? Vol. 17:23-46 (Volume publication date January 2022). https://doi.org/10.1146/annurev-pathol-042420-094031
7. Lukasiak A, Zajac M. The Distribution and Role of the CFTR Protein in the Intracellular Compartments. Membranes (Basel). 2021 Oct 22;11(11):804. doi: 10.3390/membranes11110804. PMID: 34832033; PMCID: PMC8618639.
8. Shah VS, Rajagopal J. Cystic Fibrosis: "Ionocyte Modulators"? Am J Respir Cell Mol Biol. 2023 Sep;69(3):250-252. doi: 10.1165/rcmb.2023-0169ED. PMID: 37315655; PMCID: PMC10503302.
9. Martinovich, K. M., Iosifidis, T., Buckley, A. G., Looi, K., Ling, K.-M., Sutanto, E. N., Kicic-Starcevich, E., Garratt, L. W., Shaw, N. C., Montgomery, S., Lannigan, F. J., Knight, D. A., Kicic, A., & Stick, S. M. (2017). Conditionally reprogrammed primary airway epithelial cells maintain morphology, lineage and disease specific functional characteristics. Scientific Reports, 7(1). https://doi.org/10.1038/s41598-017-17952-4 
10. GSE150674. NCBI Gene Expression Omnibus. Retrieved from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE150674
11. Mall, M. A., Criner, G. J., Miravitlles, M., Rowe, S. M., Vogelmeier, C. F., Rowlands, D. J., Schoenberger, M., & Altman, P. (2022). Cystic fibrosis transmembrane conductance regulator in COPD: A role in respiratory epithelium and beyond. European Respiratory Journal, 61(4), 2201307. https://doi.org/10.1183/13993003.01307-2022 
12. Street, K., Risso, D., Fletcher, R. et al. Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics. BMC Genomics 19, 477 (2018). https://doi.org/10.1186/s12864-018-4772-0
13. Zhicheng Ji, Hongkai Ji, TSCAN: Pseudo-time reconstruction and evaluation in single-cell RNA-seq analysis, Nucleic Acids Research, Volume 44, Issue 13, 27 July 2016, Page e117, https://doi.org/10.1093/nar/gkw430
14. Korotkevich G, Sukhov V, Sergushichev A (2019). “Fast gene set enrichment analysis.” bioRxiv. doi:10.1101/060012, http://biorxiv.org/content/early/2016/06/20/060012.
15. Liberzon, A., Birger, C., Thorvaldsdóttir, H., Ghandi, M., Mesirov, J. P., & Tamayo, P. (2015). The Molecular Signatures Database (MSigDB) hallmark gene set collection. Cell systems, 1(6), 417–425. https://doi.org/10.1016/j.cels.2015.12.004
16. Dean C. H., Snelgrove R. J. (2018). New rules for club development: new insights into human small airway epithelial club cell ontogeny and function. Am. J. Respir. Crit. Care Med. 198, 1355–1356. doi:10.1164/rccm.201805-0925ED
17. Ruobing Wang et al, De Novo Generation of Pulmonary Ionocytes from Normal and Cystic Fibrosis Human Induced Pluripotent Stem Cells, American Journal of Respiratory and Critical Care Medicine (2023). DOI: 10.1164/rccm.202205-1010LE
18. GeneCards – the human gene database www.genecards.org Stelzer G, Rosen R, Plaschkes I, Zimmerman S, Twik M, Fishilevich S, Iny Stein T, Nudel R, Lieder I, Mazor Y, Kaplan S, Dahary D, Warshawsky D, Guan - Golan Y, Kohn A, Rappaport N, Safran M, and Lancet D. The 19. GeneCards Suite: From Gene Data Mining to Disease Genome Sequence Analysis , Current Protocols in Bioinformatics(2016), 54:1.30.1 - 1.30.33.doi: 10.1002 / cpbi.5.
20. Purushothaman, A. K., & Nelson, E. J. R. (2023). Role of innate immunity and systemic inflammation in cystic fibrosis disease progression. Heliyon, 9(7), e17553. https://doi.org/10.1016/j.heliyon.2023.e17553
21. Yuan, F., Gasser, G.N., Lemire, E. et al. Transgenic ferret models define pulmonary ionocyte diversity and function. Nature 621, 857–867 (2023). https://doi.org/10.1038/s41586-023-06549-9




